### PR TITLE
Add localised sitemap generation

### PR DESCRIPTION
**Pull request by nikhilhenry**
#### What kind of changes does this PR include?

- [ ] Minor content fixes (broken links, typos, etc.)
- [ ] New or updated content
- [ ] Translated content
- [ ] Changes to the docs site code
- [x] Something else!

#### Description

- What does this PR change? Give us a brief description.
  This PR adds support for i18n sitemap generation by defining the locale config for the sitemap package to fix [issue #812](https://github.com/withastro/docs/issues/812)
- Did you change something visual? A before/after screenshot can be helpful.
   No visual changes have been made in this PR.

### CODE REVIEW

File: astro.config.ts [hunk 1]
```diff
@@ -36,7 +36,37 @@ const createSROnlyLabel = (text: string) => {
...
+				locales: {
+					ar: 'ar',
+					bg: 'bg',
+					bn: 'bn',
+					da: 'da',
+					de: 'de',
+					en: 'en-US',
+					es: 'es-ES',
+					fi: 'fi',
+					fr: 'fr-CA',
```
Comment (sarah11918) [inline astro.config.ts:54]: Are we assuming Canadian French for the docs? I know some who wouldn't mind...

File: astro.config.ts [hunk 2]
```diff
@@ -36,7 +36,37 @@ const createSROnlyLabel = (text: string) => {
...
+					it: 'it',
+					ja: 'ja',
+					kr: 'kr',
+					nl: 'nl',
+					pl: 'pl',
+					'pt-BR': 'br-PT',
+					ru: 'ru',
+					'zh-CN': 'zh-CN',
+					'zh-TW': 'zh-TW',
+				},
```
Comment (delucis) [inline astro.config.ts:65]: Sarah is right, but I think we can actually avoid duplicating this data here completely.

Instead, we can import our existing dictionary of languages:

```js
import languages from './src/i18n/languages';
```

And then generate the `locales` object from that:

```suggestion
				locales: Object.fromEntries(Object.keys(languages).map((lang) => [lang, lang])),
```

**Review by delucis — CHANGES_REQUESTED** [review]
Hi @nikhilhenry!

Thanks so much for stepping up to contribute this — really appreciate it 🙌 

I have one suggestion to simplify keeping our languages data in sync with elsewhere in the site.

Looks like there are also some conflicts now with the main branch. Could you resolve those? Let me know if you need a hand.

Thanks again 😄

**Comment by sarah11918** [comment]
Sarah may be right, but she sure wasn't very friendly...

Sorry, @nikhilhenry , I was juggling so many PRs at once, and I knew @delucis would be the one to mostly evaluate this. I just wanted to get some quick thoughts out for him.

Please don't interpret my short comments as anything more than that! It's so amazing that you showed up and started contributing to docs so quickly. We ALL recognize, and appreciate that! Can't wait until your GitHub avatar is in our FacePile on docs.  😄

**Comment by nikhilhenry** [comment]
Hi @sarah11918. Don't worry about it! I'm super glad for having been given the opportunity to help out in any way possible :)

Thank you for your support and encouragement! It would truly be an honour to have my avatar on the face pile.

I will start implementing the changes suggested by @delucis. I wasn't aware we had a dictionary. This implementation would so much more efficient!

**Comment by delucis** [comment]
@hippotastic Could you take a look at my changes to the link checker in [0290979](https://github.com/withastro/docs/pull/822/commits/029097997ee78305a916e46b8d4a01cb194eaca9)?

The new release of `@astrojs/sitemap` builds to `sitemap-0.xml` etc. instead of `sitemap.xml`. My change intends to handle any future splitting of our sitemap.

**Comment by hippotastic** [comment]
> @hippotastic Could you take a look at my changes to the link checker in [0290979](https://github.com/withastro/docs/pull/822/commits/029097997ee78305a916e46b8d4a01cb194eaca9)?
> 
> The new release of `@astrojs/sitemap` builds to `sitemap-0.xml` etc. instead of `sitemap.xml`. My change intends to handle any future splitting of our sitemap.

As already mentioned on Discord, I was returning an array from `getPagePathnamesFromSitemap()` before, now it's returning a Set. Not sure if this can cause issues further down the line, so I would prefer destructuring the set to an array again before returning it. Other than that, LGTM!

**Review by delucis — APPROVED** [review]
@nikhilhenry Thanks again for this! Welcome to the Astro docs contributors 🥳

**Comment by nikhilhenry** [comment]
@delucis and @sarah11918 thanks a ton for helping me out and guiding me throughout the process! Hope to create more PRs in the future 🚀 🚀

**Pull request merged**